typegen/src/tests/mod.rs

```diff
@@ -33,58 +33,100 @@ mod utils;
 // // 1 == 2 and 1 == 3 (both could be represented by `Foo<T,U,V>(T,U,V)`) but 2 != 3.
 // // needs more thought.
 // //
-// #[test]
-// fn more_than_1_generic_parameters() {
-//     #[allow(unused)]
-//     #[derive(TypeInfo)]
-//     struct Foo<T, U, V, W> {
-//         a: T,
-//         b: U,
-//         c: V,
-//         d: W,
-//     }
-//
-//     #[allow(unused)]
-//     #[derive(TypeInfo)]
-//     struct Bar {
-//         p: Foo<u32, u32, u64, u128>,
-//         q: Foo<u8, u8, u8, u8>,
-//     }
-//
-//     let generated = Testgen::new()
-//         .with::<Bar>()
-//         .gen_tests_mod(Default::default());
-//
-//     #[rustfmt::skip]
-//     let expected = quote!(
-//         pub mod tests {
-//             use super::types;
-//             pub struct Bar {
-//                 pub p: types::scale_typegen::tests::Foo<
-//                     ::core::primitive::u32,
-//                     ::core::primitive::u32,
-//                     ::core::primitive::u64,
-//                     ::core::primitive::u128
-//                 >,
-//                 pub q: types::scale_typegen::tests::Foo<
-//                     ::core::primitive::u8,
-//                     ::core::primitive::u8,
-//                     ::core::primitive::u8,
-//                     ::core::primitive::u8
-//                 >,
-//             }
-//             pub struct Foo<_0, _1, _2, _3> {
-//                 pub a: _0,
-//                 pub b: _1,
-//                 pub c: _2,
-//                 pub d: _3,
-//             }
-//         }
-//     );
-//
-//     assert_eq!(generated.to_string(), expected.to_string());
-// }
 
+#[test]
+fn more_than_1_generic_parameters() {
+    #[allow(unused)]
+    #[derive(TypeInfo)]
+    struct Foo<T, U, V, W> {
+        a: T,
+        b: U,
+        c: V,
+        d: W,
+    }
+
+    #[allow(unused)]
+    #[derive(TypeInfo)]
+    struct Bar {
+        p: Foo<u32, u32, u64, u128>,
+        q: Foo<u8, u8, u8, u8>,
+    }
+
+    let generated = Testgen::new()
+        .with::<Bar>()
+        .gen_tests_mod(Default::default());
+
+    #[rustfmt::skip]
+    let expected = quote!(
+        pub mod tests {
+            use super::types;
+            pub struct Bar {
+                pub p: types::scale_typegen::tests::Foo<
+                    ::core::primitive::u32,
+                    ::core::primitive::u32,
+                    ::core::primitive::u64,
+                    ::core::primitive::u128
```

Comment:
hmm ok, the codegen omits the trailing comma whereas rustfmt doesn't

Reply:
is it a problem?

Reply:
Nah, I ran into the same thing :) these quote tests are more just sanity checks than anything that we get back sensible ish output!